src/conv/invokers/impl_gemm_dynamic.cpp

```diff
@@ -304,7 +304,16 @@ InvokerFactory MakeImplGemmDynamicForward1x1InvokerFactory(const ConvolutionCont
 InvokerFactory MakeImplGemmDynamicBackwardDataInvokerFactory(const ConvolutionContext& ctx)
 {
     const auto& conv_problem = ctx.conv_problem;
-    return [conv_problem](const std::vector<Kernel>& kernels) {
+    int stride_h             = conv_problem.GetInHeight() > 1 ? conv_problem.GetKernelStrideH() : 1;
```

Comment:
Is GetInHeight() input (Hi) or really just output image height (Ho)? I know ctx swap them internally.

Ho should be used here for adjusting stride_h.

I think you need some comment here to explain if Hi/Wo get swapped, and that you are adjust stride_h based on Ho, etc